colibri/src/main/scala/colibri/Observable.scala

```diff
@@ -1086,34 +1045,86 @@ object Observable    {
 
   @inline implicit class ListSubjectOperations[A](val handler: Subject[Seq[A]]) extends AnyVal {
     def sequence: Observable[Seq[Subject.Value[A]]] = new Observable[Seq[Subject.Value[A]]] {
-      def subscribe(sink: Observer[Seq[Subject.Value[A]]]): Cancelable = {
-        handler.subscribe(Observer.create(
+      def unsafeSubscribe(sink: Observer[Seq[Subject.Value[A]]]): Cancelable = {
+        handler.unsafeSubscribe(Observer.create(
           { sequence =>
-            sink.onNext(sequence.zipWithIndex.map { case (a, idx) =>
+            sink.unsafeOnNext(sequence.zipWithIndex.map { case (a, idx) =>
               new Observer[A] with Observable.Value[A] {
                 def now(): A = a
 
-                def subscribe(sink: Observer[A]): Cancelable = {
-                  sink.onNext(a)
+                def unsafeSubscribe(sink: Observer[A]): Cancelable = {
+                  sink.unsafeOnNext(a)
                   Cancelable.empty
                 }
 
-                def onNext(value: A): Unit = {
-                  handler.onNext(sequence.updated(idx, value))
+                def unsafeOnNext(value: A): Unit = {
+                  handler.unsafeOnNext(sequence.updated(idx, value))
                 }
 
-                def onError(error: Throwable): Unit = {
-                  sink.onError(error)
+                def unsafeOnError(error: Throwable): Unit = {
+                  sink.unsafeOnError(error)
                 }
               }
             })
           },
-          sink.onError
+          sink.unsafeOnError
         ))
       }
     }
   }
 
-  private def recovered[T](action: => Unit, onError: Throwable => Unit) = try action
-  catch { case NonFatal(t) => onError(t) }
+  private def recovered[T](action: => Unit, unsafeOnError: Throwable => Unit) = try action
+  catch { case NonFatal(t) => unsafeOnError(t) }
+}
+
+private object EffectRunner {
+  def effectToIO[F[_], A](effect: F[A])(dispatcher: Dispatcher[F]): IO[A] = IO.async { cb =>
+    Async[IO].executionContext.flatMap { implicit ec =>
+      IO {
+        val (future, unsafeCancel) = dispatcher.unsafeToFutureCancelable(effect)
+        future.onComplete(t => cb(t.toEither))
+        Some(IO.fromFuture(IO(unsafeCancel())))
+      }
+    }
+  }
+
+  def unsafeRunIO[T](effect: IO[T], sink: Observer[T])(onResult: => Unit): Cancelable = {
+    import cats.effect.unsafe.{IORuntime, IORuntimeConfig}
+    import scala.concurrent.ExecutionContext
+
+    // copied from cats-effect: Dispatcher.scala
+    // this is safe because the only invocation will be cb
+    val parasitic: ExecutionContext = new ExecutionContext {
```

Comment:
Is this actually safe? It does what I want when testing it, but I am not sure.
If I use the default execution context, then we will never get a sync result from an `Observable.fromIO(IO(1))` - eventhough that was the behaviour in cats-effect 2. It seems to automatically shift when running in cats-effect 3.
I would like to not shift the initial execution - kind of using the parasitic execution context only for initial run.

Reply:
We can use `IO#syncStep` to achieve exactly that :)